cosmos/providers/dbt/core/utils/adapted_subprocesshook.py

```diff
@@ -0,0 +1,120 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import contextlib
+import os
+import signal
+from collections import namedtuple
+from subprocess import PIPE, STDOUT, Popen
+from tempfile import TemporaryDirectory, gettempdir
+
+from airflow.hooks.base import BaseHook
+
+SubprocessResult = namedtuple(
+    "SubprocessResult", ["exit_code", "output", "full_output"]
+)
+
+
+class SubprocessHook(BaseHook):
```

Comment:
What's the intent of redefining the hook here? Why can't we just subclass Airflow's hook and only make modifications where we need to?

Reply:
I have added the **full_output** method to the result, so I can fetch all the logs as a variable and carry out the analysis of the log. I thought about making the needed modifications only where it was needed, but the method **run_command**, the one I changed, consists of 90% of the code basically, so instead of subclassing the hook and creating an almost-identical copy of it, I have decided to just copy the hook itself since more lines are cheap but brainpower is usually expensive

Reply:
done